doc/motion_planning_pipeline/src/motion_planning_pipeline_tutorial.cpp

```diff
@@ -63,19 +65,43 @@ int main(int argc, char** argv)
-  robot_model::RobotModelPtr robot_model = robot_model_loader.getModel();
-
-  // Using the :moveit_core:`RobotModel`, we can construct a
-  // :planning_scene:`PlanningScene` that maintains the state of
-  // the world (including the robot).
-  planning_scene::PlanningScenePtr planning_scene(new planning_scene::PlanningScene(robot_model));
-
-  // We can now setup the PlanningPipeline
-  // object, which will use the ROS parameter server
-  // to determine the set of request adapters and the
-  // planning plugin to use
+  //     http://docs.ros.org/melodic/api/moveit_ros_planning/html/classrobot__model__loader_1_1RobotModelLoader.html
+  robot_model_loader::RobotModelLoaderPtr robot_model_loader(
+      new robot_model_loader::RobotModelLoader("robot_description"));
+
+  // Using the RobotModelLoader, we can construct a planing scene monitor that
+  // will create a planning scene, monitors planning scene diffs, and apply the diffs to it's
+  // internal planning scene. We then call startSceneMonitor, startWorldGeometryMonitor and
+  // startStateMonitor to fully initialize the planning scene monitor
+  //
+  planning_scene_monitor::PlanningSceneMonitorPtr psm(
+      new planning_scene_monitor::PlanningSceneMonitor(robot_model_loader));
+
+  /* statSceneMonitor: This method starts a subscriber which listens for planning scene messages and updates
+                       the internal planning scene accordingly */
+  psm->startSceneMonitor();
+  /* startWorldGeometryMonitor: This method starts the world geometry update monitor which listens for changes to
+                                world geometry, collision objects and optionally octomaps */
+  psm->startWorldGeometryMonitor();
+  /* startStateMonitor: Starts subscribers that listen for changes in joint state, and attached collision objects topics
+                        and update the internal planning scene accordingly*/
+  psm->startStateMonitor();
+
+  /* We can also use the RobotModelLoader to get a robot model which contains the robot's kinematic information */
+  robot_model::RobotModelPtr robot_model = robot_model_loader->getModel();
+
+  /* We can get the most up to date robot state from the PlanningSceneMonitor by locking the internal planning scene
+     for reading. This lock ensures that the underlying scene isn't updated while we are reading it's state.
+     RobotState's are useful for computing the forward and inverse kinematics of the robot among many other uses */
+  robot_state::RobotStatePtr robot_state(
+      new robot_state::RobotState(planning_scene_monitor::LockedPlanningSceneRO(psm)->getCurrentState()));
+
+  /* Create a JointModelGroup to keep track of the current robot pose and planning group. The Joint Model
+     group is useful for dealing with one set of joints at a time such as a left arm or a end effector */
+  const robot_model::JointModelGroup* joint_model_group = robot_state->getJointModelGroup("panda_arm");
```

Comment:
Please use `robot_model` to access the JMG. I consider it bad practice to use a RobotState to access stuff from the underlying RobotModel.